lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py

```diff
@@ -228,38 +421,49 @@ def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(
         dict(
-            az=dict(default=None, required=False),
+            az=dict(default='', required=False),
             cidr=dict(default=None, required=True),
+            ipv6cidr=dict(default='', required=False),
             state=dict(default='present', choices=['present', 'absent']),
             tags=dict(default={}, required=False, type='dict', aliases=['resource_tags']),
             vpc_id=dict(default=None, required=True),
-            map_public=dict(default=False, required=False, type='bool')
+            map_public=dict(default=False, required=False, type='bool'),
+            assign_ipv6_on_create=dict(default=False, required=False, type='bool')
         )
     )
 
     module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
 
+    if module.params.get('assign_ipv6_on_create') and not module.params.get('ipv6cidr'):
```

Comment:
This should probably just be `required_if`

Reply:
I initially attempted to get this working using required_if but could not and I could not find an example of using required_if to test against a boolean parameter. i.e. What I need is: `if assign_ipv6_on_create is True then ipv6cidr is a required value`

I tried this per something similar I found in one of the Azure modules but it did not work.
```python
    required_if = [('assign_ipv6_on_create', True, ['ipv6cidr'])]

    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True, required_if=required_if)
```

Reply:
Nevermind I figured this one out.

```python
required_if = [('assign_ipv6_on_create', True, ('ipv6cidr', ''))]
```